### DISCUSSION TITLE
Auto attribute to detect part needing 2D drawing

### DISCUSSION THREAD

**Discussion by bdepi23** [cwapi3d/cwapi3dpython#202] [Q&A]
Hello all
I wish to use auto attributes to detect the part which have not 8 right angles (marked as "TAILLE") such as we make 2D drawings (same as docking saw key function). it will enables to us to detect in the export file the part which has not been detected by the docking saw function or the ones which has been created after using this function (it happened a lot in the past)

I created a python script but it seems there is a bug in my cadwork file because some parts with 4 right angles are detected as "TAILLE".

Can you help me to understand this bug ?

FYI, all the part are picked from our internal catalog.
I made some tests and by creating new beans and copying attribute from the bugged parts and i have not anymore the issue. So aybe it comes from my catalog but I would like to check.

here is the link to reach the file 
https://drive.google.com/drive/folders/1kAIxq5AXgh9RPa9KcMgI1mdKivLr3wSK?usp=sharing

Many thanks for your help and have a good day

**Comment by weber-cw** [comment]
Hi,
I unfortunately won´t be able to help with the python-issue, but would like to focus on the docking-saw function. If the function gives wrong results or does not find parts at all, please additionally contact your local support with sample elements! It can´t be your task to write python control-functions for cadwork "on-board"-functions!
Regards, Matthias (cw-Hildesheim)

**Comment by Brunner246** [comment]
Hi,

I recommend that you execute your code via IDLE Shell and test it with a few elements. For example, you can use print statements to output logging statements to the console. This allows you to check whether the code is behaving correctly.

This allows you to check whether your code is working correctly. 

https://docs.cadwork.com/projects/cwapi3dpython/en/latest/get_started/

<img width="1133" alt="image" src="https://github.com/user-attachments/assets/22e2537a-4794-4522-963b-10d48c0c0619">

You could even use the IDLE for debugging (it is not the most beautiful view 🙈 ).


<img width="1102" alt="image" src="https://github.com/user-attachments/assets/bb01a316-b33d-4784-b904-9cf21a347723">
